specs/AppNotifications/AppNotificationContentSpec/AppNotificationBuilder-spec.md

```diff
@@ -0,0 +1,1201 @@
+        Green, // The button will be red
+    };
+
+    runtimeclass AppNotificationButton
+    {
+        // Button can use ToolTip instead of content which requires empty content.
+        // Argument attribute is required and uses the ArgumentSerializer construct
+        // to add arguments to the button.
+        AppNotificationButton();
+        AppNotificationButton(String buttonText);
+
+        AppNotificationButton AddArgument(String key, String value);
+
+        // Sets the IconUri for the button.
+        AppNotificationButton SetIcon(Windows.Foundation.Uri value);
+
+        // The tooltip for a button, if the button has an empty content string.
+        AppNotificationButton SetToolTip(String value);
+
+        // Sets the Button as context menu action.
+        AppNotificationButton UseContextMenuPlacement();
+
+        // Sets the ButtonColor to Green or Red
+        AppNotificationButton SetButtonColor(ButtonColor value);
+
+        // Sets the ID of an existing TextBox in order to have this button display to the right of the input.
+        AppNotificationButton SetInputId(String value);
+
+        // Launches the URI passed into the button when activated.
+        AppNotificationButton SetInvokeUri(Windows.Foundation.Uri protocolUri);
+        AppNotificationButton SetInvokeUri(Windows.Foundation.Uri protocolUri, String targetAppId);
+
+        // Retrieves the XML content of the button.
+        String GetXml();
+    };
+
+    runtimeclass AppNotificationComboBox
+    {
+        AppNotificationComboBox(String id);
+
+        // Add a selection item to the AppNotificationComboBox if InputType is AppNotificationComboBox. The id parameter is used to retrieve the user input when the app is activated
+        // and content is the displayed text for the item.
+        AppNotificationComboBox AddItem(String id, String content);
+
+        // Adds a title to display on top
+        AppNotificationComboBox SetTitle(String value);
+
+        // Sets the default selection to be displayed by the AppNotificationComboBox
+        AppNotificationComboBox SetSelectedItem(String id);
+
+        // Retrieves the XML content of the input.
+        String GetXml();
+    };
+
+    runtimeclass AppNotificationProgressBar
+    {
+        // AppNotificationProgressBar binds to AppNotificationProgressData so the AppNotification will
+        // receive every update to the status and value. In the WinAppSDK, these binding
+        // values are static, so developers won't need to define these binding values
+        // themselves.
+        AppNotificationProgressBar();
+
+        // Adds a static title to the progress bar
+        AppNotificationProgressBar UseStaticTitleForLabel(String value);
+
+        // Adds the title binding value to be used with AppNotificationProgressData
+        AppNotificationProgressBar UseBoundTitleForLabel();
+
+        // Adds the valueStringOverride value.
+        AppNotificationProgressBar UseValueStringOverride();
```

Comment:
I find this name confusing. I'm assuming this is for binding, but the naming doesn't convey that. Above you use "Bound" in the title label to convey that.

Reply:
I agree these APIs don't correctly imply they are data binding.

Would do you think about:
- BindTitleForLabel
- BindValueStringOverride
?

Reply:
That's better. Also I don't think `static` is really needed in `UseStaticTitleForLabel`. Could just be `UseTitleForLabel(...)` (static is implied by the constant you enter). I do however also get really confused between `UseStaticTitleForLabel` and `UseTitle`. They sound similar but it's not clear from the xml version how these differ.